src/matching/matching_interface.jl

```diff
@@ -0,0 +1,193 @@
+# Matching is part of an extendable interface. However, as we have not been able to
+# create more matchers than just the `MatchBySSSetDistance`, we do not expose this
+# interface to the users. Perhaps in the future we will expose this!
+
+# For now, the only parts exposed are these functions:
+export replacement_map, replacement_map!, match_sequentially!
+# all of which take as input the treshold and distance of the
+# `MatchBySSSetDistance` matcher.
+
+"""
+    IDMatcher
+
+Supertype of all "matchers" that match can IDs labelling attractors.
+
+Matchers implement an extendable interface.
+For simple matcher, one only needs to extend the function [`replacement_map`](@ref).
+This function is used by the higher level function [`match_sequentially!`](@ref).
+Matchers that may have more involved behavior requiring the dynamical system and
+current and previous parameter need to extend the internal function
+[`_match_attractors`](@ref). This function is utilized by [`AttractorSeedContinueMatch`](@ref).
+
+As you user you typically only care about given an instance of `IDMatcher`
+to a global_continuation algorithm such as [`AttractorSeedContinueMatch`](@ref),
+and you don't have to worry about the matching functions themselves.
+"""
+abstract type IDMatcher end
+
+# lowest level function API that falls back to `replacement_map`
+"""
+    _match_attractors(
+        current_attractors, prev_attractors, matcher,
+        ds, p, pprev
+    )
+
+Return the replacement map, mapping IDs of current attractors to the ones
```

Comment:
```suggestion
Returns the replacement map, mapping IDs of current attractors to the ones
```

Reply:
Julia style guide is that docstrings are written in imperative form: "do this, give that".